static/js/message_fetch.js

```diff
@@ -161,17 +166,28 @@ exports.load_messages = function (opts) {
         data.narrow = JSON.stringify(page_params.narrow);
     }
 
+    let update_loading_indicator = opts.msg_list === current_msg_list;
     if (opts.num_before > 0) {
-        opts.msg_list.data.fetch_status.start_older_batch();
+        opts.msg_list.data.fetch_status.start_older_batch({
+            update_loading_indicator: update_loading_indicator,
+        });
         if (opts.msg_list === home_msg_list) {
-            message_list.all.data.fetch_status.start_older_batch();
+            message_list.all.data.fetch_status.start_older_batch({
+                update_loading_indicator: update_loading_indicator,
```

Comment:
I think this should always be false; `message_list.all` is never rendered.  I'll fix this and add some explanatory comments that will help make this code more readable.